aws/resource_aws_gamelift_game_session_queue.go

```diff
@@ -0,0 +1,204 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/gamelift"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+)
+
+func resourceAwsGameliftGameSessionQueue() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGameliftGameSessionQueueCreate,
+		Read:   resourceAwsGameliftGameSessionQueueRead,
+		Update: resourceAwsGameliftGameSessionQueueUpdate,
+		Delete: resourceAwsGameliftGameSessionQueueDelete,
+
+		Schema: map[string]*schema.Schema{
+			"destinations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"player_latency_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"maximum_individual_player_latency_milliseconds": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+						},
+						"policy_duration_seconds": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+						},
+					},
+				},
+			},
+			"timeout_in_seconds": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(10, 600),
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGameliftGameSessionQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	input := gamelift.CreateGameSessionQueueInput{
+		Name:                  aws.String(d.Get("name").(string)),
+		Destinations:          expandGameliftGameSessionQueueDestinations(d.Get("destinations").([]interface{})),
+		PlayerLatencyPolicies: expandGameliftGameSessionPlayerLatencyPolicies(d.Get("player_latency_policy").([]interface{})),
+		TimeoutInSeconds:      aws.Int64(int64(d.Get("timeout_in_seconds").(int))),
+	}
+	log.Printf("[INFO] Creating Gamelift Session Queue: %s", input)
+	out, err := conn.CreateGameSessionQueue(&input)
+	if err != nil {
+		return err
```

Comment:
Nit: We should return context about when errors occur for operators and code maintainers, e.g.

```suggestion
		return fmt.Errorf("error creating Gamelift Game Session Queue: %s", err)
```

Reply:
:+1: